parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java

```diff
@@ -32,4 +35,21 @@ public final class OracleDALStatementVisitor extends OracleStatementVisitor impl
     public ASTNode visitAlterResourceCost(final AlterResourceCostContext ctx) {
         return new OracleAlterResourceCostStatement();
     }
+    
+    @Override
+    public ASTNode visitExplain(final ExplainContext ctx) {
+        OracleExplainStatement result = new OracleExplainStatement();
+        OracleDMLStatementVisitor visitor = new OracleDMLStatementVisitor();
+        visitor.setParameterMarkerSegments(getParameterMarkerSegments());
+        if (null != ctx.insert()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.insert()));
+        } else if (null != ctx.delete()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.delete()));
+        } else if (null != ctx.update()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.update()));
+        } else if (null != ctx.select()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.select()));
+        }
+        return result;
```

Comment:
May be you should call `result.addParameterMarkerSegments(getParameterMarkerSegments());` before return result.

Reply:
Why?

Reply:
These parameter segments are the content of this sql, if not set, the parsed sql will lose parameters.